### DISCUSSION TITLE
Comparison to Dexie.js?

### DISCUSSION THREAD

**Discussion by arahansen** [jlongster/absurd-sql#44] [Q&A]
So I'm _very_ new to the IndexDB space and am just getting caught up with what the landscape looks like. I came across a seemingly similar project, [Dexie](https://dexie.org/) that says:

> Dexie has near-native performance. Its bulk operations utilize an often-overlooked feature in IndexedDB, ignoring success callbacks when possible.

In terms of performance, is absud-sql using the same approaches as Dexie (batching operations)? What would the performance comparison for each solution look like?

**Comment by franky47** [comment]
I'm using Dexie in production, here are a few roadblocks I've hit which could make me look into something like absurd-sql:
- IndexedDB cannot do complex queries (this has little to do with Dexie): you can filter on one index, but if you want to narrow your query down (ex: _"give me items where the timestamp index is between those dates **AND** this column has this value"_), the **AND** part will have to be done in JavaScript. That means items are all pulled in memory and iterated upon using runtime code, as opposed to an SQL query with sqlite. An example: https://golb.hplar.ch/2018/01/Using-Dexie-js-in-a-TypeScript-application.html
- As mentioned in the blog post: _"you can’t even add a new “object store”, which is sort of like a table, at any point in time. You can only do it when opening the database, and doing so forces all other tabs to kill their database connection!"_.
  I thought about alleviating the first issue (no complex queries) by fragmenting my database into multiple dynamic stores, but the static tables limitation goes against it.

**Reply by tantaman** [reply]
Did you end up switching to absurd-sql or still using dexie?

**Reply by franky47** [reply]
Still using Dexie for now, the project is not at a state where data retrieval performance is the true bottleneck (it is only in stress tests).
